dmscripts/helpers/auth_helpers.py

```diff
@@ -17,12 +26,48 @@ def get_auth_token(api, stage):
     if not auth_token:
         DM_CREDENTIALS_REPO = os.environ.get('DM_CREDENTIALS_REPO', '../digitalmarketplace-credentials')
         token_prefix = 'J' if subprocess.check_output(["whoami"]) == 'jenkins' else 'D'
-        creds = subprocess.check_output([
-            "{}/sops-wrapper".format(DM_CREDENTIALS_REPO),
-            "-d",
-            "{}/vars/{}.yaml".format(DM_CREDENTIALS_REPO, stage)
-        ])
-        auth_tokens = yaml.safe_load(creds)[api]['auth_tokens']
+        creds_json = _decrypt_yaml_file_with_sops(DM_CREDENTIALS_REPO, "vars/{}.yaml".format(stage))
+        auth_tokens = creds_json[api]['auth_tokens']
         auth_token = next(token for token in auth_tokens if token.startswith(token_prefix))
 
     return auth_token
+
+
+def _get_nested_value(creds_json, env_var_path_list):
+    """
+    Handle nested values in the credentials json.
+    :param creds_json: e.g. {"performance_platform_bearer_tokens": {"g-cloud": "myToken"}}
+    :param env_var_path_list: e.g. ["performance_platform_bearer_tokens", "g-cloud"]
+    :return: e.g. "myToken"
+    """
+    if len(env_var_path_list) > 1:
+        return _get_nested_value(creds_json.get(env_var_path_list[0], {}), env_var_path_list[1:])
```

Comment:
We'll make a functional programmer of you yet... λ

Reply:
Let's rewrite the entire marketplace in Haskell.